device/sycl/src/finding/combinatorial_kalman_filter_algorithm_constant_field_default_detector.sycl

```diff
@@ -0,0 +1,60 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "../utils/get_queue.hpp"
+#include "find_tracks.hpp"
+#include "traccc/sycl/finding/combinatorial_kalman_filter_algorithm.hpp"
+
+// Detray include(s).
+#include <detray/detectors/bfield.hpp>
+#include <detray/navigation/navigator.hpp>
+#include <detray/propagator/propagator.hpp>
+#include <detray/propagator/rk_stepper.hpp>
+
+namespace traccc::sycl {
+namespace kernels::ckf_cfield_defdet {
+
+struct make_barcode_sequence;
+struct apply_interaction;
+struct find_tracks;
+struct fill_sort_keys;
+struct propagate_to_next_surface;
+struct build_tracks;
+struct prune_tracks;
+
+struct kernels {
+    using make_barcode_sequence_kernel_type = make_barcode_sequence;
+    using apply_interaction_kernel_type = apply_interaction;
+    using find_tracks_kernel_type = find_tracks;
+    using fill_sort_keys_kernel_type = fill_sort_keys;
+    using propagate_to_next_surface_kernel_type = propagate_to_next_surface;
+    using build_tracks_kernel_type = build_tracks;
+    using prune_tracks_kernel_type = prune_tracks;
+};  // namespace kernels
```

Comment:
This is not the right way to approach this problem. Instead, we should have a set of structures templated - defined not for any specific detector - on the detector type, the navigator type, the stepper type, etc. and specialize those at the kernel call sites.

Reply:
Not sure that you understand what these are for. :thinking: These types are here to allow SYCL to assign user-readable identifiers/names to the kernels that we launch.

Now... The types ("names") given to `::sycl::handler::parallel_for(...)` have to do a couple of things:
  - They can't be templated;
  - They need to be PoDs. At least they must be trivially constructible if I remember correctly.

So... None of the existing types that we use, qualify. :frowning: That's the first thing that I tried. If we want to see understandable names for the kernels in our profiles, this was the only good way I found of doing it. By defining a bunch of simple structs, all with unique names.

Reply:
:confused: I know that I ran into some sort of issue with the types that we template track finding on. It might be the trivial constructability then. But I was not able to hide this from the API of the `find_tracks` function. :frowning:

It's also an option that we say that we're okay with not-super-friendly kernel names in VTune / NSys / etc. I was seriously considering that as well. Because with that none of this would be needed.

Reply:
Yeah I think that's also an option. It's just a lot of boilerplate to have to define aliases for all of these types for each of the different detector types that we want to support, I think it would be better to avoid that. Admittedly I don't know if you are allowed to use the same type for different kernels, I assume that would cause some kind of name collision.

We can alternatively declare the structs for the `find_tracks` and other kernels independently of the detector type, and then specialize them with a label struct signifying the detector type (`defdet`, etc.).

Reply:
:thinking: Let me give that a try. Indeed, as long as the sort of templating that's happening in `contiguous_on.hpp` is accepted, there would be a better way of generating unique names here...

Reply:
It was indeed possible to make this look a whole lot nicer.

Reply:
Fantastic! Thanks! :smile: